node-manager/src/nodemanager.rs

```diff
@@ -98,6 +119,59 @@ impl NodeManager {
         self.wallet.sync().await.expect("Wallet failed to sync")
     }
 
+    #[wasm_bindgen]
+    pub async fn new_node(&self) -> String {
+        // Begin with a mutex lock so that nothing else can
+        // save or alter the node list while it is about to
+        // be saved.
+        let mut node_mutex = self.node_storage.lock().await;
+
+        // Get the current nodes and their bip32 indices
+        // so that we can create another node with the next.
+        // Always get it from our storage, the node_mutex is
+        // mostly for read only and locking.
+        let mut existing_nodes =
+            MutinyBrowserStorage::get_nodes().expect("could not retrieve nodes");
+        let next_node_index = match existing_nodes.nodes.iter().max_by_key(|n| n.child_index) {
+            None => 1,
```

Comment:
something you said about the private key of the parent exposing all children downstream of it, so i just started at one instead of 0 since most wallet will be something like `m/0/.....` right? I can start at zero either way is fine.

Reply:
Ah I see, we should probably using a longer path, at least something like `m/0'/x'` because our wallet utxos are on `86` so it'll eventually have overlap. Normally for HD paths it is 5 items long, not sure if that is required here, but we shouldn't use the the top level key.

Reply:
Using `m/0'/x'` now, thanks